### PR TITLE
Widget Test: SummaryCard

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-67

### Description

- vs Last Month" Text Validation
- Percentage Text Validation
- Amount Display Validation
- Header Text and Background Color Validation

### Screenshots

summary_card_test

### Tips

Any tips for testing it?

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [ ] All items in the author checklist is checked or not applicable
- [ ] All files including .lock and assets are reviewed
- [ ] Reasonable UI flows/cases related to the change are manually tested
- [ ] No commented-out code or unnecessary debug statements.
- [ ] No spelling or grammatical errors in code comments or documentation.
- [ ] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [ ] The branch is up-to-date with the target branch (e.g., staging).
- [ ] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: test/widgets/cards/summary_preview_card_test.dart [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                  amount: amount,
+                  percentage: percentage,
+                ),
+              ),
+            ),
+          );
+
+      testWidgets(
+        'Given SummaryPreviewCard with HeaderType.income '
+        'Then it should display "Income" and primary background color',
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart]: add When statements

File: test/widgets/cards/summary_preview_card_test.dart [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+          ));
+
+          expect(find.text('Income'), findsOneWidget);
+
+          final headerContainer = tester.widget<Container>(
+            find.descendant(
+                of: find.byType(SummaryPreviewCard),
+                matching: find.byType(Container).last),
+          );
+
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart:60]: also expect that 10 is rendered as +10% in primary color

File: test/widgets/cards/summary_preview_card_test.dart [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          // Assert
+          expect(find.text('Expense'), findsOneWidget);
+          final container = tester.widget<Container>(find
+              .descendant(
+                  of: find.byType(SummaryPreviewCard),
+                  matching: find.byType(
+                    Container,
+                  ))
+              .last);
+
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart:100]: also expect +5% is rendered with secondary color

File: test/widgets/cards/summary_preview_card_test.dart [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        },
+      );
+
+      testWidgets(
+        'Given SummaryPreviewCard '
+        'Then it should display the correct amount with ₹ symbol',
+        (WidgetTester tester) async {
+          // Arrange
+          const amount = '5000';
+          const percentage = '10';
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart]: give it -10 and expect it renders -10% is secondary color

File: test/widgets/cards/summary_preview_card_test.dart [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+          await tester.pumpWidget(createWidgetUnderTest(
+            header: HeaderType.income,
+            amount: amount,
+            percentage: percentage,
+          ));
+
+          // Assert
+          expect(find.text('₹5000'), findsOneWidget);
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart]: ₹5,000 should be expected

File: test/widgets/cards/summary_preview_card_test.dart [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        },
+      );
+
+      testWidgets(
+        'Given SummaryPreviewCard '
+        'Then it should display the correct percentage &"vs last month" text',
+        (WidgetTester tester) async {
+          // Arrange
+          const amount = '5000';
+          const percentage = '10';
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart]: send -10 and expect -10% is rendered in primary color

File: test/widgets/cards/summary_preview_card_test.dart [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+          await tester.pumpWidget(createWidgetUnderTest(
+            header: HeaderType.income,
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart]: change to headertype.expense

File: test/widgets/cards/summary_preview_card_test.dart [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+            amount: amount,
+            percentage: percentage,
+          ));
+
+          final textWidgets =
+              tester.widgetList<Text>(find.byType(Text)).toList();
+
+          expect(textWidgets[1].data, contains('₹$amount'));
```
Comment (quiteachaos) [inline test/widgets/cards/summary_preview_card_test.dart]: ₹5,000 should be expected not ₹5000.
also the test case is about rendering the vs. last month; so expect that text is rendered properly too

**Review by quiteachaos — CHANGES_REQUESTED** [review]

File: test/widgets/cards/summary_preview_card_test.dart (hunk 1 above)
Comment (krushnakantanayak) [reply test/widgets/cards/summary_preview_card_test.dart]: done

**Review by quiteachaos — APPROVED** [review]

**Pull request merged**